frontend/src/metabase/dashboard/reducers-typed.ts

```diff
@@ -413,3 +408,48 @@ export const loadingDashCards = createReducer(
       }));
   },
 );
+
+export const dashcardData = createReducer(
+  INITIAL_DASHBOARD_STATE.dashcardData,
+  builder => {
+    builder
+      .addCase(initialize, (state, action) => {
+        const { clearCache = true } = action.payload ?? {};
+        return clearCache ? {} : state;
+      })
+      .addCase(fetchCardDataAction.fulfilled, (state, action) => {
+        const { dashcard_id, card_id, result } = action.payload ?? {};
+        if (dashcard_id && card_id) {
+          return assocIn(state, [dashcard_id, card_id], result);
+        }
+      })
+      .addCase(clearCardData, (state, action) => {
+        const { cardId, dashcardId } = action.payload;
+        return dissocIn(state, [dashcardId, cardId]);
+      })
+      .addCase<string, { type: string; payload: { object: Card } }>(
+        Questions.actionTypes.UPDATE,
+        (state, action) => {
+          const id = action.payload.object.id;
+          for (const dashcardId in state) {
+            if (id in state[dashcardId]) {
+              delete state[dashcardId][id];
+            }
```

Comment:
I tested that we could delete non-existing key, would this make things a bit simpler?

```suggestion
            delete state[dashcardId][id];
```

Reply:
perfect 😄